[SOS] [Linux] Initial support of reading source and line number by native offset

lucenticus opened the pull request:
This pull request enables reading source name and line numbers from portable pdb by native offset.
 Now sos commands which are using function `GetLineByOffset` (i.e. `clru, clrstack, ip2md, pe`) show line numbers and source name. This pull request is related with `System.Diagnostics.Debug.SymbolReader.dll` from corefx PR https://github.com/dotnet/corefx/pull/9571
Related issue: https://github.com/dotnet/coreclr/issues/5920
CC @Dmitri-Botcharnikov @chunseoklee @seanshpark @mikem8361

adityamandaleeka commented on src/ToolBox/SOS/Strike/util.cpp:
```diff
@@ -6291,6 +6295,40 @@ HRESULT SymbolReader::LoadCoreCLR()
+    {
+        return Status;
+    }
+    BSTR wszFileName = SysAllocStringLen(0, cbFileName);
```
Check for null?

adityamandaleeka commented on src/ToolBox/SOS/Strike/util.cpp:
```diff
@@ -6674,6 +6712,30 @@ WString GetFrameFromAddress(TADDR frameAddr, IXCLRDataStackWalk *pStackWalk, BOO
+    if (SUCCEEDED(g_sos->GetMethodDescPtrFromIP(ip, &mdesc)))
+    {
+        DacpMethodDescData mdescData;
+        DacpModuleData dmd;
```
How about 'moduleData' instead of 'dmd'?

adityamandaleeka commented on src/ToolBox/SOS/Strike/util.cpp:
```diff
@@ -5900,7 +5902,10 @@ GetLineByOffset(
-
+#else
+        const String &moduleName = ModuleNameFromIP(TO_CDADDR(Offset));
+        return SymbolReader::GetLineByILOffset(moduleName.c_str(), MethodToken, MethodOffs, pLinenum, lpszFileName, cbFileName);
```
Does the FEEFEE stuff not apply to this scenario?

mikem8361 replied:
Hopefully the FEEFEE sequence points are“filtered” out on the managed side. I don’t remember if it was though.

lucenticus commented:
@adityamandaleeka I've fixed code, thanks for review! Please note that I've added 0xFEEFEE processing  in the commit https://github.com/dotnet/corefx/pull/9571/commits/a5972e807612c49a61b4039f8100f3932eaf4883

mikem8361 commented:
LGTM. Is this ready to be merged?

lucenticus commented:
I fixed all comments, you can merge it if @adityamandaleeka has no objection

adityamandaleeka commented:
LGTM, thanks for updating.
